ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp

```diff
@@ -694,4 +703,79 @@ bool clipboard_has_text()
   return value;
 }
 
+void clipboard_set_sprite(int ind, unsigned subimg) {
+  unsigned char *src = nullptr;
+  unsigned width, height;
+
+  enigma::sprite *spr;
+  if (!enigma::get_sprite_mtx(spr, ind)) return;  
+  src = enigma::graphics_copy_texture_pixels(spr->texturearray[subimg], &width, &height);
+
+  HBITMAP hBitmap = CreateBitmap(width, height, 1, 32, src);
+
+  OpenClipboard(NULL);
+  EmptyClipboard();
+  SetClipboardData(CF_BITMAP, hBitmap);
+  CloseClipboard();
+
+  CloseHandle(hBitmap);
+  free(src);
+}
+
+int clipboard_get_sprite(bool precise, bool transparent, bool smooth, int x_offset, int y_offset) {
+  if (!clipboard_has_sprite()) return -1;
+  OpenClipboard(NULL);
+  HBITMAP hBitmap = (HBITMAP)GetClipboardData(CF_BITMAP);
+  CloseClipboard();
+
+  BITMAPINFOHEADER bih = {0};
+  bih.biSize = sizeof(BITMAPINFOHEADER);
+
+  HDC hdc = GetDC(NULL);
+  GetDIBits(hdc, hBitmap, 0, (UINT)bih.biHeight, NULL, (BITMAPINFO *)&bih, DIB_RGB_COLORS);
+  if (bih.biBitCount <= 8) GetDIBits(hdc, hBitmap, 0, (UINT)bih.biHeight, NULL, (BITMAPINFO *)&bih, DIB_RGB_COLORS);
+
+  unsigned char *src = (unsigned char *)malloc(bih.biWidth * bih.biHeight * 4);
+  memset(src, 0, bih.biWidth * bih.biHeight * 4);
+
+  GetDIBits(hdc, hBitmap, 0, (UINT)bih.biHeight, src, (BITMAPINFO *)&bih, DIB_RGB_COLORS);
+  ReleaseDC(NULL, hdc);
+
+  std::vector<unsigned char> dst;
+  int n = bih.biWidth * bih.biHeight * 4;
+  dst.resize(n);
+
+  int i = 0;
+  for (int y = bih.biHeight - 1; y >= 0; y--) {
+    for (int x = 0; x < bih.biWidth; x++) {
+      int base = (y * bih.biWidth + x) * 4;
+      if (src[base + 3] == 0) i++;
+    }
+  }
+
+  int j = 0;
+  for (int y = bih.biHeight - 1; y >= 0; y--) {
+    for (int x = 0; x < bih.biWidth; x++) {
+      int base = (y * bih.biWidth + x) * 4;
+      dst[j++] = src[base];
+      dst[j++] = src[base + 1];
+      dst[j++] = src[base + 2];
+      dst[j++] = (i != n / 4) ? src[base + 3] : 255;
+    }
+  }
+
+  enigma::spritestructarray_reallocate();
+  enigma::sprite *spr = enigma::spritestructarray[enigma::sprite_idmax] = new enigma::sprite();
```

Comment:
Don't rely on sprite internals; they are likely to change. Use `sprite_create_empty` (or whatever it was called) to get a new sprite, then just feed it the translated bitmap data.